src/app/components-list.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ComponentLink, PLAYGROUND_COMPONENTS } from './playground-components';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ComponentsListSearchService {
+  public activeElementIndex$ = new BehaviorSubject<number>(0);
+  public inputSearch$ = new BehaviorSubject<string>('');
+  public flatFilteredComponentLinkList$ = new BehaviorSubject<ComponentLink[]>([]);
+  public selectedElement$ = new BehaviorSubject<string | any[]>('');
+
+  public componentsList$: Observable<ComponentLink[]> = this.inputSearch$.pipe(
```

Comment:
If members must be public, then remove access modifiers. Otherwise, make it private.